### PR TITLE
Add GraphQL support

### PR DESCRIPTION
**Pull request by shailscript**
Add GraphQL to project. It handles mutations and calls command handlers.
A dummy query for project is live which would be updated with a MongoDB repository once we have repository structure carved out for read models.

### CODE REVIEW

File: apps/web/src/app/app.module.ts [hunk 1]
```diff
@@ -16,6 +18,14 @@ function getProviderConfig<T>(instance: T): { provide: string, useValue: T } {
 
 @Module({
   controllers: [AppController],
+  providers: [ProjectsResolver],
+  imports: [
+    GraphQLModule.forRoot({
+      installSubscriptionHandlers: true,
+      autoSchemaFile: 'schema.gql',
```
Comment (venusvavadiya) [inline apps/web/src/app/app.module.ts:25]: Why are we generating file instead of doing it in-memory?

Comment (shailscript) [reply apps/web/src/app/app.module.ts:25]: I wasn't aware of the in-memory schema generation. Pushing an update with the requested changes.

File: apps/web/src/app/projects.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import {
+  Args, Mutation, Query, Resolver,
+} from '@nestjs/graphql';
+import {
+  CreateProjectCommand,
+  CreateProjectCommandHandler, RenameProjectCommand,
```
Comment (venusvavadiya) [inline apps/web/src/app/projects.resolver.ts:6]:
```suggestion
  CreateProjectCommandHandler,
  RenameProjectCommand,
```

File: apps/web/src/app/projects.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  RenameProjectCommandHandler,
+} from '@simple-jira/domain-project';
+import { Project } from './read-models/project.model';
+
+/* eslint-disable no-unused-vars */
+/* eslint-disable class-methods-use-this */
```
Comment (venusvavadiya) [inline apps/web/src/app/projects.resolver.ts:12]:
```suggestion
```

File: apps/web/src/app/projects.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    return project;
+  }
+
+  @Mutation((returns) => String)
+  async addProject(@Args('id') id: string) {
+    await this.createProjectCommandHandler.handle(new CreateProjectCommand(id));
+    return id;
+  }
+
+  @Mutation((returns) => String)
```
Comment (venusvavadiya) [inline apps/web/src/app/projects.resolver.ts:35]:
```suggestion
  @Mutation((returns) => string)
```

File: apps/web/src/app/projects.resolver.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Query((returns) => Project)
+  project(): any {
+    const project = new Project();
+    project.id = 'abc';
+    project.name = 'some other name';
+    return project;
+  }
+
+  @Mutation((returns) => String)
```
Comment (venusvavadiya) [inline apps/web/src/app/projects.resolver.ts:29]:
```suggestion
  @Mutation((returns) => string)
```

File: apps/web/src/app/projects.resolver.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    private readonly renameProjectCommandHandler: RenameProjectCommandHandler,
+  ) {}
+
+  @Query((returns) => Project)
+  project(): any {
+    const project = new Project();
+    project.id = 'abc';
+    project.name = 'some other name';
+    return project;
+  }
```
Comment (venusvavadiya) [inline apps/web/src/app/projects.resolver.ts:27]:
```suggestion
```

Don't need query in this PR.

File: apps/web/src/app/projects.resolver.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import {
+  Args, Mutation, Query, Resolver,
+} from '@nestjs/graphql';
```
Comment (venusvavadiya) [inline apps/web/src/app/projects.resolver.ts:3]:
```suggestion
import { Args, Mutation, Resolver } from '@nestjs/graphql';
```

File: apps/web/src/app/app.module.ts [hunk 8]
```diff
@@ -16,6 +18,13 @@ function getProviderConfig<T>(instance: T): { provide: string, useValue: T } {
 
 @Module({
   controllers: [AppController],
+  providers: [ProjectsResolver],
+  imports: [
+    GraphQLModule.forRoot({
+      autoSchemaFile: true,
+      path: 'projects',
```
Comment (venusvavadiya) [inline apps/web/src/app/app.module.ts:25]: What is the default value for this? We should not be using `projects` here.

**Review by venusvavadiya — CHANGES_REQUESTED** [review]

File: apps/web/src/app/app.module.ts (hunk 8 above)
Comment (shailscript) [reply apps/web/src/app/app.module.ts:25]: Default is `/graphql`

Comment (shailscript) [reply apps/web/src/app/app.module.ts]: Updated to be default, [reference](https://graphql.org/learn/serving-over-http/).

**Review by venusvavadiya — CHANGES_REQUESTED** [review]

File: apps/web/src/app/app.module.ts [hunk 9]
```diff
@@ -16,6 +18,12 @@ function getProviderConfig<T>(instance: T): { provide: string, useValue: T } {
 
 @Module({
   controllers: [AppController],
+  providers: [ProjectsResolver],
+  imports: [
+    GraphQLModule.forRoot({
+      autoSchemaFile: true,
```
[the remark this answers is not in the document]
Comment (shailscript) [reply apps/web/src/app/app.module.ts]: It isn't set by default.
Error: `Apollo Server requires either an existing schema, modules or typeDefs`

**Pull request closed without merging**